### DISCUSSION TITLE
Module Federation v2 examples: are they outdated

### DISCUSSION THREAD

**Discussion by abelta** [web-infra-dev/rsbuild#3911] [Q&A]
Regarding Module Federation v2, there's this line in the example:

`import { ModuleFederationPlugin } from '@module-federation/enhanced/rspack';`

however the [@module-federation/enhanced](https://github.com/module-federation/enhanced) library hasn't been updated in two years.

I'm facing a very specific problem where I need to list the host application as remote for state sharing purposes and the application fails as soon as I introduce this change.

```
name: 'host',
filename: 'remoteEntry.js',
remotes: {
  host: 'host@http://localhost:5000/remoteEntry.js',
  remote1: 'remote1@http://localhost:5001/remoteEntry.js',
},
...
```

this is however the recommended way to do it in all of the tutorials I've found so far.
I've spent weeks trying to trace the issue and only recently realised the library might be outdated.
Could this be it?

**Comment by chenjiahan** [comment]
The `@module-federation/enhanced` package is actively maintained.

You can find the source code or report issues here: https://github.com/module-federation/core

**Comment by YanPes** [comment]
@abelta remove the "filename". Since you use the default, it is not needed.  
In addition,  use "mf-manifest.json" instead of "remoteEntry.js" in your remote http url string. This is the recommended way with 2.0. Please let me know if this helped.
